lando-pushlog.rst

```diff
@@ -0,0 +1,221 @@
+   clients when they have exhausted all known pushes.
+
+.. _push_objects:
+
+Push Objects
+^^^^^^^^^^^^
+
+The value of each entry in the ``pushes`` object is an object describing
+the push and the commits therein.
+
+The following properties are always present:
+
+``commits``
+   An array of commit entries.
+
+   By default, entries are 40 character commit SHA-1s included in the
+   push. If ``full=1`` is specified, entries are objects containing
+   commit metadata (see `Commit objects <commit_objects_>`_ below).
+
+.. todo::
+
+  ENFORCE THIS
+
+   Commits are in DAG/revlog order with the tip-most commit last.
+
+``date``
+   Integer seconds since UNIX epoch that the push occurred.
+
+   For pushes that take a very long time (more than a single second),
+   the data will be recorded towards the end of the push, just before
+   the transaction is committed to Mercurial. Although, this is an
+   implementation details.
+
+   There is no guarantee of strict ordering between dates. i.e. the
+   ``date`` of push ID ``N + 1`` could be less than the ``date`` of push
+   ID ``N``. Such is how clocks work.
+
+user
+   The string username that performed the push.
+
+.. _commit_objects:
+
+Commit objects
+^^^^^^^^^^^^^^
+
+If ``full=1`` is specified, each entry in the ``commits`` array will be an
+object instead of a string. Each object will have the following properties:
+
+``commit``
+   The 40 byte hex SHA-1 of the commit.
+
+``parents``
+   An array of 1 or 2 elements containing the 40 byte hex SHA-1 of the
+   parent commit. Merges have 2 entries.
+
+.. todo::
+
+  CONFIRM THIS
+   Root changesets have the
+   value ``0000000000000000000000000000000000000000``.
+
+``author``
+   The author string from the changeset.
+
+``desc``
+   The changeset's commit message.
+
+``branch``
+   The branch the changeset belongs to.
+
+``files``
+   An array of filenames that were changed by this changeset.
+
+Here's an example::
+
+  {
+    "author": "Eugen Sawin <esawin@mozilla.com>",
+    "desc": "Bug 1110212 - Strong randomness for Android DNS resolver. r=sworkman",
+    "files": [
+      "other-licenses/android/res_init.c"
+    ],
+    "commit": "ee4fe2ec168e719e822dabcdd797c0cff9ce2407",
+    "parents": [
+      "803bc910c45a875d9d76dc689c45dd91a1e02e23"
+    ]
+  }
+
+
+
+***************
+PushLog webview
+***************
+
+.. todo::
+
+  URL `/pushlog`
+
+.. todo::
+
+  Screenshot
+
+.. _pulse_notifications:
+
+*******************
+Pulse notifications
+*******************
+
+Whenever a change has landed to a PushLog-enabled repository, Lando will send a Pulse notification.
+
+A single message can represent any number of pushes to various `branches`. It may also contain any number of `tags`.
+
+.. note:: For the purpose of syncing changes from Git to Mercuria, the `commit_id` in each in the `tags` object needs to exist in the target repository prior to the tag being created. It may be present in the `branches` object as part of the same message.
```

Comment:
```suggestion
.. note:: For the purpose of syncing changes from Git to Mercurial, the `commit_id` in each in the `tags` object needs to exist in the target repository prior to the tag being created. It may be present in the `branches` object as part of the same message.
```